packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -118,7 +118,7 @@ class AppContext {
     if (value == null && _parent != null) {
       value = _parent!.get<T>();
     }
-    return _unboxNull(value ?? _generateIfNecessary(T, _fallbacks)) as T;
+    return _unboxNull(value ?? _generateIfNecessary(T, _fallbacks)) as T?;
```

Comment:
Also in https://github.com/flutter/flutter/pull/79985, let's see which one merges first.